rwengine/src/engine/SaveGame.cpp

```diff
@@ -533,17 +533,12 @@ bool SaveGame::loadGame(GameState& state, const std::string& file)
 	for(;; bytes++ ) {
 		if(state.basic.saveName[bytes-1] == 0 && state.basic.saveName[bytes] == 0) break;
 	}
-	size_t outSize = 24;
+	size_t outSize = 0;
 	char outBuff[48];
-	char* outCur = outBuff;
-	auto icv = iconv_open("UTF-8", "UTF-16");
-	char* saveName = (char*)state.basic.saveName;
-
-#if defined(RW_NETBSD)
-	iconv(icv, (const char**)&saveName, &bytes, &outCur, &outSize);
-#else
-	iconv(icv, &saveName, &bytes, &outCur, &outSize);
-#endif
+	wchar_t* saveName = (wchar_t*)state.basic.saveName;
+
+	outSize = std::wcstombs(outBuff, saveName, sizeof(outBuff));
+	state.basic.saveName[outSize] = '\0';
```

Comment:
Possible crash if conversion fails. Manual null termination may be not necessary here, because source array is always null terminated?